core/sync/theil_sen.go

```diff
@@ -0,0 +1,102 @@
+package sync
+
+import (
+	"sort"
+	"time"
+
+	"go.uber.org/zap"
+
+	"example.com/scion-time/base/timebase"
+)
+
+type theilSen struct {
+	log        *zap.Logger
+	clk        timebase.LocalClock
+	datapoints []point
+}
+
+func newTheilSen(log *zap.Logger, clk timebase.LocalClock) *theilSen {
+	return &theilSen{log: log, clk: clk, datapoints: make([]point, 64)}
+}
+
+type point struct {
+	x float64
+	y float64
+}
+
+func sortAndGetMedian(data []float64) float64 {
+	var length = len(data)
+	if length == 0 {
+		panic("invalid argument: array is empty, median undefined")
+	}
+
+	sort.Float64s(data)
+
+	if length%2 == 0 {
+		return (data[length/2-1] + data[length/2]) / 2
```

Comment:
I've now tried to align the two implementations as much as possible. I'm a bit unsure why `timemath.Median` defines `m`, assigns to it and then returns it instead of directly returning the values. Is there a reason for this?

Reply:
Ah, that's just to have a single exit of the function in the success case. Should make it a simpler to reason about and provide a single point to add postcondition checks, for example.